src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -412,6 +455,50 @@ export class SearchSource {
           const dataFrameResponse = response as IDataFrameResponse;
           await this.setDataFrame(dataFrameResponse.body as IDataFrame);
           return onResponse(searchRequest, convertResult(response as IDataFrameResponse));
+        } else if ((response as IDataFrameResponse).type === DATA_FRAME_TYPES.POLLING) {
+          const dataFrameResponse = response as IDataFrameResponse;
+          const responseQueryId = (dataFrameResponse.body as IDataFrame).meta?.queryId;
+          const responseSessionId = (dataFrameResponse.body as IDataFrame).meta?.sessionId;
+          // if there wasn't a session before, set current datasource as session
+          // TODO: MQL update with selected datasource
```

Comment:
i know it probably isnt the best supported with the data source being specific for mys3 but i think we should be able to use dataframe.name as right now that is the datasource. and then for demo purposes we can modify the read only advanced setting